kolibri/core/auth/management/commands/generate_auth_data.py

```diff
@@ -0,0 +1,663 @@
+import csv
+import datetime
+import os
+import random
+
+from django.core.management import call_command
+from django.core.management.base import BaseCommand
+from le_utils.constants import content_kinds
+
+from kolibri.core.auth.constants import demographics
+from kolibri.core.auth.constants import facility_presets
+from kolibri.core.auth.constants import role_kinds
+from kolibri.core.auth.models import AdHocGroup
+from kolibri.core.auth.models import Classroom
+from kolibri.core.auth.models import Facility
+from kolibri.core.auth.models import FacilityDataset
+from kolibri.core.auth.models import FacilityUser
+from kolibri.core.auth.models import LearnerGroup
+from kolibri.core.content.management.commands.generate_content_data import (
+    generate_channels,
+)
+from kolibri.core.content.management.commands.generate_content_data import (
+    switch_to_memory,
+)
+from kolibri.core.content.models import ChannelMetadata
+from kolibri.core.content.models import ContentNode
+from kolibri.core.exams.models import Exam
+from kolibri.core.exams.models import ExamAssignment
+from kolibri.core.lessons.models import Lesson
+from kolibri.core.lessons.models import LessonAssignment
+from kolibri.core.utils.csv import open_csv_for_reading
+from kolibri.utils.time_utils import local_now
+
+users_data_iterator = 0
+all_users_base_data = []
+
+
+def generate_random_id():
+    import uuid
+
+    return uuid.uuid4().hex
+
+
+def read_user_data_file(file_path):
+
+    if not file_path:
+        file_path = os.path.abspath(
+            os.path.join(
+                os.path.dirname(__file__),
+                "..",
+                "..",
+                "..",
+                "logger",
+                "management",
+                "commands",
+                "user_data.csv",
+            )
+        )
+
+    global all_users_base_data
+    file_reader = open_csv_for_reading(file_path)
+    all_users_base_data = [data for data in csv.DictReader(file_reader)]
+
+
+# ever time gets called it gets the next (different) user data from user_data.csv
+def get_user_base_data(n_facility_users):
+    global users_data_iterator
```

Comment:
when you modify (and not just access) a global variable (variables outisde function are global by default) from inside a function you have to specity the keyword global
and it's global because this makes since to be iterated every time relevant function gets called

Reply:
also if you noticed ` users_data_iterator` is accessed from 2 functions, so global makese sense right?